### PR TITLE
feature(build) Switches from TravisCI to GitHub actions

### LINKED ISSUE #379: Migrate CI/CD to GitHub Actions
As described in  accordproject/technical-steering-committee#4
   
   Please follow the conventions in https://github.com/accordproject/concerto/pull/236
   
   - [x] Remove secrets from Travis (for AP maintainers: in Travis Repository Settings)
   - [x] Disable Travis in the configuration (for AP maintainers: Travis Personal Profile > Settings > Accord Project
   - [x] Remove `.travis.yml` and `.travis` directories in repo
   - [x] Add `.github/workflows` directory based on those in `./concerto/.github/workflows`
   - [x] ~Remove scripts in `./scripts` replace by those found in `./concerto/scripts`~
   - [x] Fix `package.json` notable adjustments: remove some of the old scripts, ~make sure code coverage works~
   - [x] ~Test build on PR with timestamps~
   - [x] ~Test build on GitHub release with version number~
   - [x] Update GitHub "branch protection" rules to point to the new workflows instead of Travis

### PR DESCRIPTION
**Pull request by Aniruddha-Shriwant**
### Changes

- [x] Remove `.travis.yml` in repo
- [x] Add `.github/workflows` directory based on those in `./concerto/.github/workflows`

### Author Checklist
- [x] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [ ] Vital features and changes captured in unit and/or integration tests
- [x] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [ ] Extend the documentation, if necessary
- [x] Merging to `master` from `fork:branchname`

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      - name: git checkout
+        uses: actions/checkout@v2
+
+      - name: Use Node.js 14.x
+        uses: actions/setup-node@v1
+        with:
+          node-version: 14.x
+    
+      - name: Build
+        run: npx lerna bootstrap
```
Comment (Aniruddha-Shriwant) [inline .github/workflows/build.yml:85]: Should I replace `npx lerna bootstrap` with `npm run build` ?

**Comment by Aniruddha-Shriwant** [comment]
Do we need `deploy.yml` like this:
```
name: 'Netlify Preview Deploy'

on:
  pull_request:
    types: ['opened', 'edited', 'synchronize']

jobs:
  deploy:
    name: 'Deploy'
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v1
      - uses: jsmrcaga/action-netlify-deploy@master
        with:
          NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_AUTH_TOKEN }}
          NETLIFY_SITE_ID: ${{ secrets.NETLIFY_SITE_ID }}
```
Or the Netlify App will do the work of deployment?

File: .github/workflows/build.yml (hunk 1 above)
Comment (jeromesimeon) [reply .github/workflows/build.yml:85]: this looks like the wrong script to me. The latest version should be at: https://github.com/accordproject/concerto/tree/master/.github/workflows

**Comment by jeromesimeon** [comment]
> Do we need `deploy.yml` like this:
> 
> ```
> name: 'Netlify Preview Deploy'
> 
> on:
>   pull_request:
>     types: ['opened', 'edited', 'synchronize']
> 
> jobs:
>   deploy:
>     name: 'Deploy'
>     runs-on: ubuntu-latest
> 
>     steps:
>       - uses: actions/checkout@v1
>       - uses: jsmrcaga/action-netlify-deploy@master
>         with:
>           NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_AUTH_TOKEN }}
>           NETLIFY_SITE_ID: ${{ secrets.NETLIFY_SITE_ID }}
> ```
> 
> Or the Netlify App will do the work of deployment?

I don't think we need any change for Netlify. In theory the Concerto scripts should work as is. Netlify will deploy automatically.

**Comment by Aniruddha-Shriwant** [comment]
> I don't think we need any change for Netlify. In theory the Concerto scripts should work as is. Netlify will deploy automatically.

Okay

File: .github/workflows/build.yml (hunk 1 above)
Comment (Aniruddha-Shriwant) [reply .github/workflows/build.yml:85]: Can you look the file again it is same as https://github.com/accordproject/concerto/blob/master/.github/workflows/build.yml

**Comment by jeromesimeon** [comment]
@Aniruddha-Shriwant Can you please change the issue description: this will only be addressing part of #379

**Comment by Aniruddha-Shriwant** [comment]
> @Aniruddha-Shriwant Can you please change the issue description: this will only be addressing part of #379

Is it okay now?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    steps:
+      - name: git checkout
+        uses: actions/checkout@v2
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+    
+      - run: npm ci
```
Comment (jeromesimeon) [inline .github/workflows/build.yml:34]: You will need to add a `postinstall` line to call lerna for this to work. Similar to what is in here: 

https://github.com/accordproject/concerto/blob/c2ecad3e5d52b0e8b1c7caf026d62abbf4f68b6f/package.json#L25-L26

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      - run: npm run build --if-present
+      - run: npm test
+
+      - name: Calculate code coverage
+        run: npm run coverage
```
Comment (jeromesimeon) [inline .github/workflows/build.yml:39]: There is no code coverage in this repo (so this command will fail).

I would suggest to delete the part of the workflow which handles code coverage.

**Review by jeromesimeon — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 3 above)
Comment (Aniruddha-Shriwant) [reply .github/workflows/build.yml:39]: Ohh... okay I will update it

File: scripts/timestamp.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const path = require('path');
+const semver = require('semver');
+const dayjs = require('dayjs');
```
Comment (jeromesimeon) [inline scripts/timestamp.js:20]: Sorry, but another change will be needed for this: to remove `moment-mini` from the dependencies and add `dayjs` in the `package.json`.

Comment (Aniruddha-Shriwant) [reply scripts/timestamp.js:20]: Sure! I will change this too

Comment (jeromesimeon) [reply scripts/timestamp.js:20]: Great! you can find the corresponding `dayjs` package version again in the concerto `package.json`

File: .github/workflows/build.yml (hunk 3 above)
Comment (Aniruddha-Shriwant) [reply .github/workflows/build.yml]: I had updated it can you check the code please

File: scripts/timestamp.js (hunk 4 above)
Comment (Aniruddha-Shriwant) [reply scripts/timestamp.js:20]: Ohh sorry for that I had installed latest version is that okay?
or should I install one that is there in concerto's `package.json`

Comment (Aniruddha-Shriwant) [reply scripts/timestamp.js:20]: @jeromesimeon I had added the corresponding version of `dayjs`

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+    
+      - run: npm ci
+      - run: npm run build --if-present
+      - run: npm test
+
+  publish:
```
[the remark this answers is not in the document]
Comment (Aniruddha-Shriwant) [reply .github/workflows/build.yml:38]: No problem! I was also confused about publishing to npm :)
Thanks for pointing that out I will resolve this soon...

[the remark this answers is not in the document]
Comment (Aniruddha-Shriwant) [reply .github/workflows/build.yml]: As I had removed  `moment-mini` from the dependencies, should I add it back?

[the remark this answers is not in the document]
Comment (jeromesimeon) [reply .github/workflows/build.yml]: > As I had removed `moment-mini` from the dependencies, should I add it back?

very good question! I asked myself the same thing and double check and I think it's simply an unused dependency. So ok to keep it out. Thanks @Aniruddha-Shriwant

**Review by jeromesimeon — COMMENTED** [review]
Looks great! Simple too. Thanks for the contribution.

I will do further testing once merged

**Review by jeromesimeon — APPROVED** [review]

**Pull request merged**